### PR TITLE
Adding support for intel OneAPI compiler: icx, icpx, and ifx

### PR DESCRIPTION
**Pull request by josemonsalve2**
Adding the Intel OneAPI configuration and iris.def file for ANL.

### CODE REVIEW

File: sys/make/make.def [hunk 1]
```diff
@@ -192,6 +202,16 @@ ifeq ($(CXX), xlc++)
   CXXLINKFLAGS += -O3 -qsmp=omp $(CXXOFFLOADING)
 endif
 
+# Intel ICPX compiler
+ifeq ($(CXX), icpx)
+  CXXOFFLOADING = -fopenmp-targets=spir64
+  CXX_NO_OFFLOADING =
+  CXXFLAGS += -lm -O3 -fiopenmp $(COFFLOADING) -D__STRICT_ANSI__
```
Comment (spophale) [inline sys/make/make.def:209]: Use CXXOFFLOADING. The values may be the same now, but it breaks pattern.

File: sys/make/make.def [hunk 2]
```diff
@@ -192,6 +202,16 @@ ifeq ($(CXX), xlc++)
...
+  CXXLINK = icpx
+  CXXLINKFLAGS += -lm -O3 -fiopenmp $(COFFLOADING)
```
Comment (spophale) [inline sys/make/make.def:211]: Same as above.

File: sys/make/make.def [hunk 3]
```diff
@@ -266,6 +286,16 @@ ifeq ($(FC), $(filter $(FC), xlf xlf_r))
   FLINKFLAGS += -O3 -qsmp=omp $(FOFFLOADING)
 endif
 
+# Intel IFX compiler
+ifeq ($(FC), ifx)
+  FOFFLOADING = -fopenmp-targets=spir64
+  F_NO_OFFLOADING =
+  FFLAGS += -O3 -fiopenmp $(COFFLOADING)
```
Comment (spophale) [inline sys/make/make.def:293]: Use FOFFLOADING.

File: sys/make/make.def [hunk 4]
```diff
@@ -266,6 +286,16 @@ ifeq ($(FC), $(filter $(FC), xlf xlf_r))
...
+  FLINK = ifx
+  FLINKFLAGS += -O3 -fiopenmp $(COFFLOADING)
```
Comment (spophale) [inline sys/make/make.def:295]: Same as above.

File: sys/make/make.def [hunk 5]
```diff
@@ -266,6 +286,16 @@ ifeq ($(FC), $(filter $(FC), xlf xlf_r))
...
+  F_VERSION = echo "$(shell $(call loadModules,$(C_COMPILER_MODULE),"shut up") ifx --version | grep 'ifx' | grep -oh '[0-9.]\+[a-zA-Z0-9. ]*')"
```
Comment (spophale) [inline sys/make/make.def:296]: Shouldn't it be F_COMPILER_MODULE ?

File: sys/make/make.def [hunk 6]
```diff
@@ -192,6 +202,16 @@ ifeq ($(CXX), xlc++)
...
 endif
 
+# Intel ICPX compiler
+ifeq ($(CXX), icpx)
+  CXXOFFLOADING = -fopenmp-targets=spir64
+  CXX_NO_OFFLOADING =
+  CXXFLAGS += -lm -O3 -fiopenmp $(COFFLOADING) -D__STRICT_ANSI__
+  CXXLINK = icpx
+  CXXLINKFLAGS += -lm -O3 -fiopenmp $(COFFLOADING)
+  CXX_VERSION = echo "$(shell $(call loadModules,$(C_COMPILER_MODULE),"shut up") icpx -dumpversion)"
```
Comment (spophale) [inline sys/make/make.def:212]: Shouldn't it be CXX_COMPILER_MODULE ?

**Comment by colleeneb** [comment]
This worked ok for me on JLSE! 

I get the same results as I get by doing it manually (e.g. `CC='icx -fiopenmp -fopenmp-targets=spir64'`).

 The only thing is that I might suggest is changing `iris` to `JLSE`, since `JLSE` is the system name. :)

**Review by colleeneb — APPROVED** [review]

**Comment by spophale** [comment]
@jhdavis8 or @tmh97, could either of you verify the changes so that we may merge it asap?

**Review by jhdavis8 — APPROVED** [review]
Looks good to me

**Pull request merged**